lexers/embedded/hare.xml

```diff
@@ -0,0 +1,104 @@
+<lexer>
+  <config>
+    <name>Hare</name>
+    <alias>hare</alias>
+    <filename>*.ha</filename>
+    <mime_type>text/x-hare</mime_type>
+  </config>
+  <rules>
+    <state name="whitespace">
+      <rule pattern="^use\s.+;">
+        <token type="CommentPreproc"/>
```

Comment:
I don't like how this is done, it makes the whole `use` statement a uniform and weird greenish color:
![image](https://github.com/alecthomas/chroma/assets/32234660/72e7e06f-a7f3-4e18-bfeb-5f45326e195b)

Compare to Helix, which highlights each part of the statement separately:
![image](https://github.com/alecthomas/chroma/assets/32234660/929c5585-d43d-4b28-ab34-35435394a49e)

Reply:
You can use `<bygroups>` to achieve this.

Reply:
Thanks, I decided to have `use` be a `KeywordNamespace` but not use a separate state or `bygroups` for the rest of the `use` statement, as I thought it was a lot of hassle for little benefit as it already looks fine with my change:
![image](https://github.com/alecthomas/chroma/assets/32234660/25eb91c9-25e1-4222-b1a9-6076899dde35)